packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2914,8 +2928,54 @@ class _SerializationDelegate implements DiagnosticsSerializationDelegate {
   @override
   final bool expandPropertyValues;
 
+  /// Callback to add additional experimental serialization properties.
+  ///
+  /// This callback can be used to customize the serialization of DiagnosticsNode
+  /// objects for experimental features in widget inspector clients such as
+  /// [Dart DevTools](https://github.com/flutter/devtools).
+  /// For example, [Dart DevTools](https://github.com/flutter/devtools)
+  /// can evaluate the following expression to register a VM Service API
+  /// with a custom serialization to experiment with visualizing layouts.
+  ///
+  /// The following code samples demonstrates adding the RenderObject associated
+  /// with an Element to the serialized data for all elements in the tree:
+  /// ```dart
+  /// Map<String, Object> getDetailsSubtreeWithRenderObject(
+  ///   String id,
+  ///   String groupName,
+  ///   int subtreeDepth,
+  /// ) {
+  ///   return _nodeToJson(
+  ///     root,
+  ///     InspectorSerializationDelegate(
```

Comment:
Where would I run this code? InspectorSerializationDelegate is marked as visibleForTesting, so would this only be in test code?

Reply:
Debugging tools such as Dart DevTools can access it even if it was still private via the VM Service protocol. It is marked `@visibleForTesting` so we can test it from widget_inspector_test.dart

Reply:
This code is run through the eval in the [Dart DevTools](https://github.com/flutter/devtools/pull/1396/files#diff-64bc585ceb9f869917465f175f0a9c6e). 
But currently in the flutter framework yes it is only in the test code.